ldb/admin.py

```diff
@@ -60,10 +66,126 @@ def __init__(self, *args, **kwds):
         self.fields['living_with'].queryset = Person.objects.order_by('surname', 'firstname')
 
 
+class PersonResource(resources.ModelResource):
+    def before_import(self, dataset, using_transactions, dry_run, **kwargs):
+        if 'id' not in dataset.headers:
+            dataset.lpush_col([None for row in range(len(dataset))], header='id')
+
+    def init_instance(self, row=None):
+        instance = self._meta.model()
+        try:
+            student = Student.objects.get(
+                student_number=str(row['student__student_number'])  # Cast to string as it is saved like this in the DB
+            )
+        except Student.DoesNotExist:
+            student = Student()
+
+        try:
+            member = Member.objects.get(
+                person_id=row['id']
+            )
+        except Member.DoesNotExist:
+            member = Member()
+
+        instance.member = member
+        instance.student = student
+        return instance
+
+    def get_or_init_instance(self, instance_loader, row):
+        """
+        Either fetches an already existing instance or initializes a new one.
+        """
+        try:
+            instance = Person.objects.get(
+                # These should be unique values.
+                netid=str(row['netid']),
+                # Cast to string as it is saved like this in the DB
+                student__student_number=str(row['student__student_number'])
```

Comment:
Yeah I am not sure about this. They are both unique values in the DB. However the lookup now is when both netid and student number are the same. It should actually be if one of the 2 matches. But then what if that makes 2 people match. Or when we've filled in 'unknown' because these values can not be null, but sometimes that data is missing from the data we get.

Perhaps we should make these values not required as a first step?

Reply:
Maybe this is an idea: first try to get the student by both `netid` and `student_number`, then if it is not found only get by `netid`, then try by `student_number` and only then make a new instance?

Reply:
So that seems to work, it finds the person and replaces values where possible.

![Screenshot 2019-06-27 at 14 43 10](https://user-images.githubusercontent.com/1493561/60267471-9882f800-98ea-11e9-94fc-3e52ab0fdb13.png)

Reply:
Could you check if you agree with how it works now?

Reply:
Hmmm 🤔 I'm not sure how many people we don't have netid or student number, but if it works it works. So I would say keep it this way.

Reply:
Yeah this is fine. You could change to `Person.objects.get(Q(netid=str(row['netid']) | student__student_number=str(row['student__student_number']))`, but this is also fine.

Reply:
Oh I like that much better. Will change that.